app/controllers/secondary_content_links_controller.rb

```diff
@@ -33,21 +33,43 @@ def destroy
 
       redirect_to step_by_step_page_secondary_content_links_path(step_by_step_page.id), notice: 'Secondary content link was successfully deleted.'
     else
-      redirect_to step_by_step_page_secondary_content_links_path
+      redirect_to step_by_step_page_secondary_content_links_path(step_by_step_page.id)
     end
   end
 
+  def index
+    step_by_step_page
+    secondary_content_links
+    @new_secondary_content_link = new_secondary_content_link
+  end
+
 private
 
   def step_by_step_page
     @step_by_step_page = StepByStepPage.find(params[:step_by_step_page_id])
   end
 
+  def secondary_content_links
+    @secondary_content_links = StepByStepPage.find(params[:step_by_step_page_id]).secondary_content_links
+  end
+
+  def new_secondary_content_link
+    step_by_step_page.secondary_content_links.new
```

Comment:
@DilwoarH do we need this? It's doing the same thing as the `new` action.

Reply:
I think `new` could be changed to:

```
  def new
    @secondary_content_link = step_by_step_page.secondary_content_links.new
  end
```

And the form created around `@secondary_content_link`?

Reply:
The issue is (correct me if I'm wrong) that the new would assume that there would be another view called `new.html.erb`. In our current approach, everything happens in one view, therefore, this wouldn't be needed.

Reply:
Not sure tbh. One to try out.